goi18n/goi18n.go

```diff
@@ -6,12 +6,82 @@ import (
 	"os"
 )
 
+type commander interface {
+	execute() error
+}
+
+func main() {
+	flag.Usage = usage
+
+	mergeCmd := flag.NewFlagSet("merge", flag.ExitOnError)
+	mergeCmd.Usage = usageMerge
+	sourceLanguage := mergeCmd.String("sourceLanguage", "en-us", "")
+	outdir := mergeCmd.String("outdir", ".", "")
+	format := mergeCmd.String("format", "json", "")
+
+	constantsCmd := flag.NewFlagSet("constants", flag.ExitOnError)
+	constantsCmd.Usage = usageConstants
+	packageName := constantsCmd.String("package", "R", "")
+	outdirConstants := constantsCmd.String("outdir", ".", "")
+
+	if len(os.Args) == 1 {
+		usage()
+	}
+
+	switch os.Args[1] {
+	case "merge":
+		mergeCmd.Parse(os.Args[2:])
+	case "constants":
+		constantsCmd.Parse(os.Args[2:])
+	default:
+		mergeCmd.Parse(os.Args[1:])
+	}
+
+	var cmd commander
+
+	if mergeCmd.Parsed() {
+		cmd = &mergeCommand{
+			translationFiles:  mergeCmd.Args(),
+			sourceLanguageTag: *sourceLanguage,
+			outdir:            *outdir,
+			format:            *format,
+		}
+	} else if constantsCmd.Parsed() {
+		cmd = &constantsCommand{
+			translationFiles: constantsCmd.Args(),
+			packageName:      *packageName,
+			outdir:           *outdirConstants,
+		}
+	}
+
+	if err := cmd.execute(); err != nil {
+		fmt.Println(err.Error())
+		os.Exit(1)
+	}
+}
+
 func usage() {
-	fmt.Printf(`goi18n formats and merges translation files.
+	fmt.Printf(`goi18n manages translation files.
 
 Usage:
 
-    goi18n [options] [files...]
+    goi18n merge     Merge translation files
+    goi18n constants Generate constant file from translation file
+
+For more details execute:
+
+    goi18n [command] -help
+
+`)
+	os.Exit(1)
+}
+
+func usageMerge() {
```

Comment:
Can you move this to be defined in the file where the merge command is defined? Same with `usageConstants()`